File: Command/Java/src/Commands/HottubOffCommand.java
package Commands;

import Devices.Hottub;

public class HottubOffCommand implements Command {
    Hottub hottub;

    public HottubOffCommand(Hottub hottub) {
        this.hottub = hottub;
    }

    public void execute() {
        hottub.setTemperature(36);
        hottub.off();
    }
    public void undo() {
        hottub.on();
    }
}

File: Command/TypeScript/src/commands/HottubOffCommand.ts
import Hottub from '../devices/Hottub';
import Command from './Command';

export default class HottubOffCommand implements Command {
  hottub: Hottub;

  constructor(hottub: Hottub) {
    this.hottub = hottub;
  }

  execute() {
    this.hottub.setTemperature(36);
    this.hottub.off();
  }

  undo() {
    this.hottub.on();
  }
}
